SAP/spartacus
# fix: remove unnecessary loading of checkout data

closes #8898 

QA Steps:

- Checkout to `develop` branch
- Add a product to the `cart`
- Monitor Checkout details call through redux or network tab
- `Proceed To Checkout` and continue to `Delivery Mode`
- Go to the `cart` and do any interaction like add or remove product within cart
- Expected unnecessary checkout call on every interaction
- Do the same in branch `feature/GH-8898`
